src/main/java/de/diddiz/util/MySQLConnectionPool.java

```diff
@@ -1,437 +1,39 @@
 package de.diddiz.util;
 
-import static de.diddiz.LogBlock.config.Config.mb4;
+import com.zaxxer.hikari.HikariDataSource;
+
 import java.io.Closeable;
-import java.sql.Array;
-import java.sql.Blob;
-import java.sql.CallableStatement;
-import java.sql.Clob;
 import java.sql.Connection;
-import java.sql.DatabaseMetaData;
-import java.sql.DriverManager;
-import java.sql.NClob;
-import java.sql.PreparedStatement;
-import java.sql.SQLClientInfoException;
 import java.sql.SQLException;
-import java.sql.SQLWarning;
-import java.sql.SQLXML;
-import java.sql.Savepoint;
-import java.sql.Statement;
-import java.sql.Struct;
-import java.util.Enumeration;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Properties;
-import java.util.Vector;
-import java.util.concurrent.Executor;
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
 
 public class MySQLConnectionPool implements Closeable
 {
-	private final static int poolSize = 10;
-	private final static long timeToLive = 300000;
-	private final Vector<JDCConnection> connections;
-	private final String url, user, password;
-	private final Lock lock = new ReentrantLock();
+
+	private final HikariDataSource ds;
 
 	public MySQLConnectionPool(String url, String user, String password) throws ClassNotFoundException {
-		Class.forName("com.mysql.jdbc.Driver");
-		this.url = url;
-		this.user = user;
-		this.password = password;
-		connections = new Vector<JDCConnection>(poolSize);
-        ConnectionReaper reaper = new ConnectionReaper();
-		new Thread(reaper, "MySQL Connection Reaper Thread - LogBlock").start();
+		this.ds = new HikariDataSource();
+		ds.setJdbcUrl(url);
+		ds.setUsername(user);
+		ds.setPassword(password);
+
+		ds.addDataSourceProperty("useUnicode", "true");
+		ds.addDataSourceProperty("characterEncoding", "utf-8");
+		ds.addDataSourceProperty("rewriteBatchedStatements", "true");
+
+		ds.addDataSourceProperty("cachePrepStmts", "true");
+		ds.addDataSourceProperty("prepStmtCacheSize", "250");
+		ds.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+		ds.addDataSourceProperty("useServerPrepStmts", "true");

```

Comment:
Good point, I think batch rewriting will probably end up on top but benchmarking is always good.